src/parsing/xpath.pegjs

```diff
@@ -121,8 +121,12 @@ RangeExpr
 
 // 21
 AdditiveExpr
- = lhs:MultiplicativeExpr _ op:("-" / "+") _ rhs:AdditiveExpr {return ["binaryOperator", op, lhs, rhs]}
- / MultiplicativeExpr
+ = lhs:MultiplicativeExpr rest:( _ op:("-" / "+") _ rhs:MultiplicativeExpr {return {op: op, rhs: rhs}})* {
+        return rest.length === 0 ? lhs : rest.reduce(function (inner, nesting) {
```

Comment:
The `length === 0` is not actually needed; Array.prototype.reduce returns the initial value if the array is empty. But let's be explicit here